File: app/migrate/tree_depth.py
# coding=utf-8
# author: Xiguang Liu<g10guang@foxmail.com>
# 2018-05-02 13:46
# 题目描述：https://www.nowcoder.com/practice/435fb86331474282a3499955f0a41e8b?tpId=13&tqId=11191&rp=1&ru=%2Fta%2Fcoding-interviews&qru=%2Fta%2Fcoding-interviews%2Fquestion-ranking


class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None


class Solution:
    def TreeDepth(self, pRoot):
        if not pRoot:
            return 0
        return max(self.TreeDepth(pRoot.left), self.TreeDepth(pRoot.right)) + 1

File: app/migrate/tree_depth.go
package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

/**
 * 代码中的类名、方法名、参数名已经指定，请勿修改，直接返回方法规定的值即可
 *
 * @param pRoot TreeNode类
 * @return int整型
 */
func TreeDepth(pRoot *TreeNode) int {
	var depthFn func(node *TreeNode) int
	depthFn = func(node *TreeNode) int {
		if node == nil {
			return 0
		}
		left := depthFn(node.Left)
		right := depthFn(node.Right)
		return max(left, right) + 1
	}
	return depthFn(pRoot)
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}
